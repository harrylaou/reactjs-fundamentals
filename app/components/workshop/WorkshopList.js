import React from "react";

const WorkshopList = () => (

    <h1>There are no courses</h1>
)

export default WorkshopList