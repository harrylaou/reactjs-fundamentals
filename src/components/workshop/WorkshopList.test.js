import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import WorkshopList from './WorkshopList'

describe('<WorkshopList />', () => {
  it('should render with default props', () => {
    const onClick = jest.fn();
    const workshop = {
      id: 'rfct123',
      title: 'React Fundamentals',
      price: 990,
      instructors: [{ avatar: 'goo.gl/7f00gI'}]
    }
    const props = {
      workshops: [workshop],
      width: 3,
      showWorkshop: onClick
    }

    const wrapper = shallow(
      <WorkshopList {...props} />
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()

    wrapper.find('ListItem').simulate('click')

    expect(onClick).toBeCalledWith(workshop)
  })
})