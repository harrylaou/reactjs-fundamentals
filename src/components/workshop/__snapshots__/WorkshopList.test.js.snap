// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkshopList /> should render with default props 1`] = `
<div
  className="view-container"
>
  <List
    className="view"
  >
    <ListItem
      autoGenerateNestedIndicator={true}
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      leftIcon={
        <ActionGrade
          color="#ff4081"
        />
      }
      nestedItems={Array []}
      nestedLevel={0}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="React Fundamentals  -  990"
      primaryTogglesNestedList={false}
      rightAvatar={
        <div>
          <Avatar
            size={40}
            src="goo.gl/7f00gI"
          />
        </div>
      }
      secondaryTextLines={1}
      style={
        Object {
          "color": "black",
        }
      }
    />
  </List>
  <div
    className="view"
  />
</div>
`;

exports[`<WorkshopList /> should render with default props 2`] = `
<div
  className="view-container"
>
  <div
    className="view"
  />
</div>
`;
